server-spi/src/main/java/org/keycloak/models/credential/PasswordCredentialModel.java

```diff
@@ -15,7 +15,7 @@
     private final PasswordCredentialData credentialData;
     private final PasswordSecretData secretData;
 
-    private PasswordCredentialModel(PasswordCredentialData credentialData, PasswordSecretData secretData) {
+    public PasswordCredentialModel(PasswordCredentialData credentialData, PasswordSecretData secretData) {
```

Comment:
Just for the sake of compatibility with other implementations, could we keep the constructor private and instead introduce other static method like for example:

```
public static PasswordCredentialModel createFromValues(String algorithm, byte[] salt, int hashIterations, String 
encodedPassword, MultivaluedHashMap<String, String> otherCredentialData,, MultivaluedHashMap<String, String> otherSecretData)
```
The current method signature "createFromValues" needs to be kept as well for the backwards compatibility, but it can just call the new method. Something like:
```
public static PasswordCredentialModel createFromValues(String algorithm, byte[] salt, int hashIterations, String encodedPassword) {
    createFromValues(algorithm, salt, hashIterations, encodedPassword, Collections.emptyMap(), Collections.emptyMap());
}
```

Reply:
@mposolda Yeah, you're right about the credential data. I was thinking about a separated and more general-purpose thing like what you mentioned before in regards to something like "otherClaims". And possibly avoid migration, etc.

But if you think we can reuse that column, that is fine.

Reply:
I agree, we can move to a static factory + private constructor.